File: py-leetcode/1535.py
class Solution:
    def getWinner(self, arr: List[int], k: int) -> int:
        n = len(arr)
        curmax, currk = arr[0], 0
        for i in range(1, n):
            if currk == k:
                return curmax
            if curmax > arr[i]:
                currk += 1
            else:
                currk = 1
                curmax = arr[i]
        return curmax

File: swift-leetcode/1535.swift
class Solution {
    func getWinner(_ arr: [Int], _ k: Int) -> Int {
        let n = arr.count;
        var currk = 0;
        var currMax = arr[0];
        for i in 1..<n {
            if currk == k {
                return currMax;
            }
            if currMax > arr[i] {
                currk+=1;
            } else {
                currk = 1;
                currMax = arr[i];
            }
        }
        return currMax;
    }
}